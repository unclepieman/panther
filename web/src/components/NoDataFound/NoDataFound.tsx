/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { Heading, Flex, Img } from 'pouncejs';
import NoDataFoundIllustration from 'Assets/illustrations/charts.svg';

interface NoDataFoundProps {
  title?: string;
  children?: never;
}

const NoDataFound: React.FC<NoDataFoundProps> = ({ title = "You don't have any data" }) => {
  return (
    <Flex height="100%" direction="column" align="center" justify="center">
      <Img nativeWidth={80} nativeHeight={90} alt="Charts" src={NoDataFoundIllustration} />
      <Heading size="x-small" color="navyblue-100" mt={6}>
        {title}
      </Heading>
    </Flex>
  );
};

export default NoDataFound;
