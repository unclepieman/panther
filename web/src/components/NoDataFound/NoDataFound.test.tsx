/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { render } from 'test-utils';
import NoDataFound from './NoDataFound';

describe('NoDataFound', () => {
  it('matches snapshot', () => {
    const { container } = render(<NoDataFound />);
    expect(container).toMatchSnapshot();
  });

  it('contains proper semantics', () => {
    const { getByText, getByAltText } = render(<NoDataFound />);
    expect(getByAltText('Charts')).toBeInTheDocument();
    expect(getByText("You don't have any data")).toBeInTheDocument();
  });

  it('allows a user to override the text', () => {
    const { getByText } = render(<NoDataFound title="Fake title" />);
    expect(getByText('Fake title')).toBeInTheDocument();
  });
});
