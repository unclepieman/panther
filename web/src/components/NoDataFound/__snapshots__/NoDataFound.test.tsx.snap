// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoDataFound matches snapshot 1`] = `
.panther-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.panther-0 {
  max-width: 100%;
}

.panther-1 {
  font-weight: 400;
  font-size: 1.125rem;
  color: #86a3c3;
  margin-top: 24px;
}

<div>
  <header
    id="main-header"
  />
  <div
    class="panther-2"
  >
    <img
      alt="Charts"
      class="panther-0"
      height="90"
      src="test-file-stub"
      width="80"
    />
    <h1
      class="panther-1"
    >
      You don't have any data
    </h1>
  </div>
  <footer
    id="footer"
  />
</div>
`;
